### ISSUE TITLE
How to get edge_index from unordered point clouds?

### ISSUE THREAD

**Issue by xiaomingjie** [rusty1s/pytorch_geometric#882]
## ❓ Questions & Help

Almost all the graph convolutions need edge_index which is provided by some certain datasets. With those datasests such as ShapeNet whose point clouds are unordered and with no edges in them, is there any approaches to get edge_index and perform such gnns?

**Comment by rusty1s** [comment]
Have a look at the `radius_graph` and `knn_graph` methods. They exist both as normal functions and as transforms.

**Comment by xiaomingjie** [comment]
Thank you for your kindly reply.  Is `data.pos` a matrix of n*3 which n indicates point numbers and 3 indicateds xyz coordinate?

**Comment by rusty1s** [comment]
Yes, but both methods do also work with any number of features (not just 3).

**Comment by xiaomingjie** [comment]
Thanks. Question solved.

**Comment by xiaomingjie** [comment]
Excuse me. When I tried to perform knn_graph, I found that I cant get data.batch in my own dataset. my data format is like [b,n,c] while knn_graph needed [n,c]. How can I perform knn_graph batch by batch?

**Comment by rusty1s** [comment]
You need to either reshape your data to [n, c] and create your batch object manually
```
torch.arange(0, b).view(-1, 1).repeat(1, n).view(-1)
```
or default to other libraries such as the [KeOps library](https://www.kernel-operations.io/keops/index.html) which might handle this regular case much faster :)

**Comment by tchaton** [comment]
Hey @xiaomingjie, have a look at https://github.com/nicolas-chaulet/deeppointcloud-benchmarks
